katran/lib/KatranMonitor.cpp

```diff
@@ -49,11 +50,18 @@ KatranMonitor::KatranMonitor(const KatranMonitorConfig& config)
     throw std::runtime_error("none of eventReaders were initialized");
   }
 
-  std::vector<std::shared_ptr<DataWriter>> data_writers;
+std::vector<std::shared_ptr<DataWriter>> data_writers;
   for (int i = 0; i < config_.maxEvents; i++) {
-    std::string fname;
-    folly::toAppend(config_.path, "_", i, &fname);
-    data_writers.push_back(std::make_shared<FileWriter>(fname));
+    if (config_.storage == PcapStorageFormat::FILE) {
+      std::string fname;
+      folly::toAppend(config_.path, "_", i, &fname);
+      data_writers.push_back(std::make_shared<FileWriter>(fname));
+    } else {
+      // PcapStorageFormat::IOBuf
+      buffers_.emplace_back(folly::IOBuf::create(config_.bufferSize));
```

Comment:
Here buffer is emplaced back to the last index. Later (in getEventBuffer) you try to access buffer with index of 'event'. What associates a buffer with an event-type?

Reply:
line 54 above. events are enumerated from 0 to up to maxEvents. eventId is an index inside data_writers and buffers (if IOBuf storage is being used)